src/Error.purs

```diff
@@ -0,0 +1,120 @@
+module Error
+  ( E
+  , NotImplementedError
+  , _notImplementedError
+  , notImplementedError
+  , traceAndHushAll
+  , traceAndHushAll_
+  , hushable
+  , safe_
+  , noteE
+  , noteEM
+  ) where
+
+import Data.Maybe
+
+import Contract.Prelude
+  ( class Monad
+  , Either(..)
+  , pure
+  , unwrap
+  , (*>)
+  , (<<<)
+  , (>>=)
+  , (>>>)
+  )
+import Control.Monad.Cont (lift)
+import Control.Monad.Error.Class (throwError)
+import Control.Monad.Except (ExceptT(..), runExceptT)
+import Control.Monad.Except.Checked (ExceptV, safe)
+import Control.Monad.Maybe.Trans (MaybeT(..), runMaybeT)
+import Data.Function (($))
+import Data.Identity (Identity)
+import Data.Variant (Variant, inj)
+import Debug (class DebugWarning, traceM)
+import Helpers (notImplemented)
+import Prim.Row (class Union)
+import Prim.TypeError (class Warn, Text)
+import Type.Proxy (Proxy(Proxy))
+import Type.Row (type (+))
+
+-- we could use the left side to store logs to get better error tracing
+type E v a = Either (Variant v) a
+
+type NotImplementedError r = (notImplementedError :: String | r)
+_notImplementedError = Proxy :: Proxy "notImplementedError"
+
+notImplementedError
+  :: forall (r :: Row Type) (a :: Type)
+   . Warn (Text "Function not implemented!")
+  => String
+  -> E (NotImplementedError + r) a
+notImplementedError = throwError <<< inj _notImplementedError
+
+-- | Allows hush errors tracing as debug outputs.
+-- | For use when error hadling is not yet supported at a call site.
+traceAndHushAll_
+  :: forall (v :: Row Type) (a :: Type)
+   . DebugWarning
+  => ExceptV v Identity a
+  -> Maybe a
+traceAndHushAll_ = unwrap <<< runMaybeT <<< traceAndHushAll
+
+-- | Allows hush errors tracing as debug outputs.
+-- | For use when error hadling is not yet supported at a call site.
+traceAndHushAll
+  :: forall (v :: Row Type) (m :: Type -> Type) (a :: Type)
+   . DebugWarning
+  => Monad m
+  => ExceptV v m a
+  -> MaybeT m a
+traceAndHushAll act = MaybeT $ runExceptT act >>= case _ of
+  Left v -> traceM v *> pure Nothing
+  Right a -> pure (pure a)
+
+-- | Lifts underlying monad to MaybeT allowing hushing selected errors.
+hushable
```

Comment:
I don't see how can it be used, could you add an example? Is it adapted from somewhere?

Reply:
I've added an example to the docstring.